src/precompilation/FrontendCompiler.h

```diff
@@ -99,9 +99,9 @@ namespace vc4c
         void assembleSPIRV(SPIRVTextSource&& source, const std::string& userOptions, SPIRVResult& result);
         void compileLLVMToSPIRVText(LLVMIRSource&& source, const std::string& userOptions, SPIRVTextResult& result);
         void disassembleSPIRV(SPIRVSource&& source, const std::string& userOptions, SPIRVTextResult& result);
-
         void linkLLVMModules(std::vector<LLVMIRSource>&& sources, const std::string& userOptions, LLVMIRResult& result);
         void linkSPIRVModules(std::vector<SPIRVSource>&& sources, const std::string& userOptions, SPIRVResult& result);
+        void optimizeByOpt(std::string& result);
```

Comment:
Also fir further improvements: this function should be eventually rewritten to match the signatures of the others to allow arbitrary input ans output (file or stream)

Reply:
Yes, it should be better. But I am not familiar with front-end code, I couldn't do it now...